test/test-server/networking_handlers.go

```diff
@@ -1890,23 +1890,36 @@ func handleNetworkingNetworkLinkServiceAssociationList(t *testing.T) http.Handle
 			err := writeErrorJson(w, "NetworkLinkEndpoints are not found")
 			require.NoError(t, err)
 		default:
-			pageToken := r.URL.Query().Get("page_token")
-			var associationList networkingv1.NetworkingV1NetworkLinkServiceAssociationList
-			switch pageToken {
-			case "nls-association-2":
-				associationList = networkingv1.NetworkingV1NetworkLinkServiceAssociationList{
-					Data:     []networkingv1.NetworkingV1NetworkLinkServiceAssociation{association2},
-					Metadata: networkingv1.ListMeta{},
-				}
-			default:
-				associationList = networkingv1.NetworkingV1NetworkLinkServiceAssociationList{
-					Data:     []networkingv1.NetworkingV1NetworkLinkServiceAssociation{association1},
-					Metadata: networkingv1.ListMeta{Next: *networkingv1.NewNullableString(networkingv1.PtrString("/networking/v1/network-link-service-associations?environment=env-00000&page_size=1&page_token=nls-association-2"))},
-				}
-			}
-
-			err := json.NewEncoder(w).Encode(associationList)
+			nlsList := getNetworkLinkServiceAssociationList(networkLinkService, phase)
+			err := json.NewEncoder(w).Encode(nlsList)
 			require.NoError(t, err)
 		}
 	}
 }
+
+func getNetworkLinkServiceAssociationList(filterNetworkLinkService string, filterPhase []string) networkingv1.NetworkingV1NetworkLinkServiceAssociationList {
+	association1 := getNetworkLinkServiceAssociation("nle-111111", getNetworkLinkEndpoint("nle-111111", ""))
+	association2 := getNetworkLinkServiceAssociation("nle-222222", getNetworkLinkEndpoint("nle-222222", "my-network-link-endpoint-2"))
+	association1.Status.SetPhase("PENDING_ACCEPT")
+
+	associationList := networkingv1.NetworkingV1NetworkLinkServiceAssociationList{
+		Data: []networkingv1.NetworkingV1NetworkLinkServiceAssociation{
+			association1, association2,
+		},
+	}
+	associationList.Data = filterAssociationList(associationList.Data, filterNetworkLinkService, filterPhase)
```

Comment:
Call this in the initialization above?

Reply:
Was following the format from [schema registry](https://github.com/confluentinc/cli/blob/0bb590ba67c1ce16fd4381ad1212bce237e099cf/test/test-server/srcm_handlers.go#L133), I don't feel strongly either way but the rest of the networking list commands follow this format.